plugins/chain_plugin/chain_plugin.cpp

```diff
@@ -229,6 +229,25 @@ void chain_plugin::set_program_options(options_description& cli, options_descrip
    cfg.add_options()
          ("blocks-dir", bpo::value<bfs::path>()->default_value("blocks"),
           "the location of the blocks directory (absolute path or relative to application data dir)")
+         ("blocks-log-stride", bpo::value<uint32_t>(),
+         "split the block log file when the head block number is the multiple of the stride\n"
+         "When the stride is reached, the current block log and index will be renamed '<blocks-retained-dir>/blocks-<start num>-<end num>.log/index'\n"
+         "and a new current block log and index will be created with the most recent block. All files following\n"
+         "this format will be used to construct an extended block log.")
+         ("max-retained-block-files", bpo::value<uint32_t>(),
+          "the maximum number of blocks files to retain so that the blocks in those files can be queried.\n"
+          "When the number is reached, the oldest block file would be moved to archive dir or deleted if the archive dir is empty.\n"
+          "The retained block log files should not be manipulated by users." )
+         ("blocks-retained-dir", bpo::value<bfs::path>(),
+          "the location of the blocks retained directory (absolute path or relative to blocks dir).\n"
+          "If the value is empty, it is set to the value of blocks dir.")
+         ("blocks-archive-dir", bpo::value<bfs::path>(),
+          "the location of the blocks archive directory (absolute path or relative to blocks dir).\n"
+          "If the value is empty, blocks files beyond the retained limit will be deleted.\n"
+          "All files in the archive directory are completely under user's control, i.e. they won't be accessed by nodeos anymore.")
+         ("fix-irreversible-blocks",
```

Comment:
This doesn't seem like something that should be a nodeos option, this looks like a leap-util thing

Reply:
leap-util has the option to fix corrupted block log content already. This is here for the operator who doesn't care about why the log is corrupted and just want to fix it and restart the nodeos anyway. Without the flag, the operator may have to write an extra shell script to detect the situation and use leap-util to correct the problem then restart the nodeos. The option provides the functionality without the complication of shell script.

Reply:
I'd like to get another opinion from at least one of @heifner @arhag or @766C6164 on this. 

While this isn't an option that performs an operation and then quits (_clearly_ leap-util), it's still an operation that sounds like some sort of offline recovery of a nodeos data file, which is in the ballpark of items we're been moving out of nodeos to leap-util.

It's also a problematic option from the standpoint of a user trying to decide whether or not to enable it.. why wouldn't I _always_ have this enabled? I searched through the entire diff and didn't find any error message that told the user when they should use the option or not.

It feels like if we detect an inconsistency in the block log, we should exit with an error message telling the user something is wrong and they should perform a recovery via leap-util. At least then they can decide what kind of recovery to perform. As is, it seems like it'll decide the recovery strategy based on whatever the option is -- an option the user probably didn't even know existed -- hope it's what they wanted!

Reply:
> It feels like if we detect an inconsistency in the block log, we should exit with an error message telling the user something is wrong and they should perform a recovery via leap-util. At least then they can decide what kind of recovery to perform. As is, it seems like it'll decide the recovery strategy based on whatever the option is -- an option the user probably didn't even know existed -- hope it's what they wanted!

I agree. I think we should exit with an error message and let the user decide recovery strategy via `leap-util` or just restore from their own backup. In many cases the user is likely to restore from there own backup as a faster way to get running again instead of `nodeos` doing some long running recovery step.

Reply:
option removed